src/hotspot/share/gc/shared/collectedHeap.cpp

```diff
@@ -548,7 +548,8 @@ void CollectedHeap::full_gc_dump(GCTimer* timer, bool before) {
     GCTraceTime(Trace, gc, classhisto) tm(before ? "Class Histogram (before full gc)" : "Class Histogram (after full gc)", timer);
     ResourceMark rm;
     LogStream ls(lt);
-    VM_GC_HeapInspection inspector(&ls, false /* ! full gc */);
+    uint parallel_thread_num = MAX2<uint>(1, (uint)os::initial_active_processor_count() * 3 / 8);
+    VM_GC_HeapInspection inspector(&ls, false /* ! full gc */, parallel_thread_num);
```

Comment:
Pre-existing: the comment next to a bool constant in a parameter list should show the name of the parameter, i.e. s/! full_gc/request_full_gc/. Maybe change this as appropriate in other callers too.

Reply:
Your call.